data-finding/src/DataReader.py

```diff
@@ -4,176 +4,121 @@
       write_file.write(",".join(menu_line) + "\n")
       for row in reader:
-        row = self.cleanLine(row)
+        row = cleanLine(row)
         if row:
           write_file.write(",".join(row) + "\n")
-          # print(row)
-    print("Complete")
+    logging.info("Complete")
     write_file.close()
+    return True
 
-
-
-class FileSplitter:
-  initPath = ""
-  file_finder = {}
-  file_writers = []
-  file_index = -1
-
-  def getWD(self, path):
-    path = path[::-1]
-    search = re.search("/", path) 
-    if search:
-      path = path[search.span()[1]:]
-      path = path[::-1]
-      print(path)
-    return path
-
-  def setInitPath(self, path):
-    self.initPath = self.getWD(path)
-  
-  #This function will split a csv into multiple files based on a certain key
-  def split(self, file_path, key):
-    
-    index_counter = 0 #counter to track writers
-    with open(file_path, newline='') as csv_file:
-      reader = csv.reader(csv_file, delimiter=',', quotechar='"')
-      #Grab the first line and find the key
-      menu = reader.__next__()
-      for i in range(len(menu)):
-        if key == menu[i]:
-          self.file_index = i
-      if self.file_index == -1:
-        print("Unable to find key in CSV file. Operation cancelled.")
-        return False 
-      #If key found, create a folder as there can be an extreme amount of CSV files
-      writer_path = file_path[0:len(file_path)-4] + "_files"
-      os.mkdir( writer_path)
-      #Run through CSV using key index to split csv into multiple files
-      for row in reader:
-        #If file not already made, create a file and write to it.
-        if row[self.file_index] not in self.file_finder:
-          self.file_finder[row[self.file_index]] = index_counter
-          index_counter += 1
-          new_file_path = writer_path + "/" + row[self.file_index] + "data.csv"
-          if os.path.exists(new_file_path):
-            os.remove(new_file_path)
-          self.file_writers.append(open(new_file_path,"x"))
-          self.file_writers[self.file_finder[row[self.file_index]]].write(",".join(menu) + "\n")
-        row[2] = "\"" + row[2] + "\""
-        self.file_writers[self.file_finder[row[self.file_index]]].write(",".join(row) + "\n")
-      #Close all files before exiting
-      for i in self.file_writers:
-        i.close()  
-    print("complete")  
-    
-def openFile():
-  reader = FileReader()
-  file_path = filedialog.askopenfilename(initialdir=reader.initPath)
+#depending on what button is pressed, this function is called to handle button presses and will process a file
+def openFile(**kwargs) -> None:
+  file_path = filedialog.askopenfilename(initialdir=initPath)
   if type(file_path) == str and file_path.endswith('.csv'):
-    reader.parseCSV(file_path)
-    reader.setInitPath(file_path)
+    logging.info("Operation completed: " + parseCSV(file_path, **kwargs))
+    setInitPath(file_path)
   elif type(file_path) == str:
     try:
       file = open(file_path,'r')
```

Comment:
you should look into the python 'with' keyword, this can help reduce your code complexity especially with the explicit file.close()

covers try/catch blocks too